### ISSUE TITLE
[Bug]: helm-index (http://helm.appsmith.com) not listing old charts / only newest chart is listed

### ISSUE THREAD

**Issue by WernerDreierB** [appsmithorg/appsmith#11521]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a deployment-pipeline using your official helm-repo pinned on a chart Version (e.g. 1.5.0)
If meanwhile a new Appsmith-Chart-Version (e.g. 1.5.1) is released, the configured helm-repo (http://helm.appsmith.com) does not list Chart Version 1.5.0 anymore, but only the newer released 1.5.1

This is not how an helm-repo chart-index should work. The helm-repo http://helm.appsmith.com should list all charts ever created as a list, not only the newest one.

See https://helm.sh/docs/topics/chart_repository/ for an example

### Steps To Reproduce

1. Download index-file from http://helm.appsmith.com
2. Verify, that only one Chart-Version exists (meaning older chart versions are inaccessible)

### Environment

Production

### Version

Self-Hosted

**Comment by sharat87** [comment]
Hey @WernerDreierB, thank you for reporting this and providing details. I've confirmed that we indeed have the chart files, but unfortunately aren't merging the `index.yaml` file. We'll work getting this fixed.

In your particular case, would it be possible to update to the latest version of the charts? Is there a reason you'd want to stay with the old version of the chart? No disagreeing with the issue here, just curious to see if there's anything else we can do to improve the experience. Thanks!

**Comment by WernerDreierB** [comment]
Hey @sharat87  thank you for getting back to me on this so quickly. 
In particular, we were able to upgrade to the newest chart, so this no blocker. It simply puts our pipeline at risk of failing somewhen in the future, as we will not neccessarily have the same upgrade shedule on our self-hosted environment as you guys proceed with releasing new charts
Thank you and thank you also for the great application you are building

**Comment by sharat87** [comment]
Ah I see. Thanks for the details (and the kind words 😄).

You've probably already realized this, but I'll just share here again. The version of the chart itself, is not tightly connected to the version of Appsmith you are running. Appsmith should be able auto-update itself even if you are running the same version of the chart. This happens at a schedule that you can configure in your chart installation. This is documented at <https://docs.appsmith.com/setup/kubernetes#auto-update-charts-image>.

**Comment by WernerDreierB** [comment]
Sure, the auto-update is a great feature for smaller teams, however as we are running our own data-center with strictly audited (and firewalled) deployment-pipelines. Therefore we cannot rely on auto-update features but have to actively do upgrades on our own.

**Comment by WernerDreierB** [comment]
Ping. 
Stumbled upon the same issue again with the newest Chart 1.5.3

**Comment by sharat87** [comment]
@WernerDreierB, sorry, but I see version 1.5.3 in the `index.yaml`, so it's there, no? Sorry if I misunderstood.

**Comment by WernerDreierB** [comment]
> @WernerDreierB, sorry, but I see version 1.5.3 in the `index.yaml`, so it's there, no? Sorry if I misunderstood.

Sure, 1.5.3 is in there... but normally index-page (https://helm.appsmith.com) should list all ever existing chart-versions, so that "old" pipelines can still run with the older version of the chart.

**Comment by sharat87** [comment]
Okay, I'm still a little lost, sorry. So are you saying that 1.5.3 is there, but the older ones like 1.5.0 aren't there? Or are you saying that 1.5.3 isn't there in the index page? Because I can see it when I do a `curl`:

```
>  curl -s https://helm.appsmith.com | grep 1.5.3
    - https://helm.appsmith.com/appsmith-1.5.3.tgz
    version: 1.5.3
```

**Comment by WernerDreierB** [comment]
> Okay, I'm still a little lost, sorry. So are you saying that 1.5.3 is there, but the older ones like 1.5.0 aren't there? Or are you saying that 1.5.3 isn't there in the index page? Because I can see it when I do a `curl`:
> 
> ```
> >  curl -s https://helm.appsmith.com | grep 1.5.3
>     - https://helm.appsmith.com/appsmith-1.5.3.tgz
>     version: 1.5.3
> ```

Yes, I am saying that the older ones like 1.5.0 aren't listed anymore and therefore are not installable anymore. The index page at https://helm.appsmith.com/ should keep the older entries (as it is a list of charts), not meant to contain only the latest existing chart

**Comment by pratapaprasanna** [comment]
@WernerDreierB  please check now the we have added old charts as well now.
